app/src/main/java/org/mozilla/focus/tiles/HomeTilesManager.kt

```diff
@@ -78,7 +81,12 @@ class BundledTilesManager private constructor(context: Context) {
     fun unpinSite(context: Context, uri: Uri): Boolean {
         for (pair in bundledTilesCache) {
             if (compareUri(uri, pair.key)) {
-                // TODO: Remove by pair.value.id (BEFORE removing from cache)
+                val blacklist = context.getSharedPreferences(PREF_HOME_TILES, MODE_PRIVATE).getStringSet(BUNDLED_SITES_BLACKLIST, mutableSetOf())
+                val edit = context.getSharedPreferences(PREF_HOME_TILES, MODE_PRIVATE).edit()
+                val newBlacklist = blacklist.toMutableSet()
+                newBlacklist.add(pair.value.id)
+                edit.putStringSet(BUNDLED_SITES_BLACKLIST, newBlacklist)
+                edit.apply()
```

Comment:
You can just put the apply at the end of the previous edit line - see documentation, all of these put commands return an instance of the editor.

https://developer.android.com/reference/android/content/SharedPreferences.Editor.html